h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -721,13 +721,23 @@ public final double likelihood(double yr, double ym) {
     public GLMWeights computeWeights(double y, double eta, double off, double w, GLMWeights x) {
       double etaOff = eta + off;
       x.mu = linkInv(etaOff);
-      double var = variance(x.mu);//Math.max(1e-5, variance(x.mu)); // avoid numerical problems with 0 variance
+      double var = variance(x.mu);
       double d = linkDeriv(x.mu);
       x.w = w / (var * d * d);
       x.z = eta + (y - x.mu) * d;
       likelihoodAndDeviance(y,x,w);
       return x;
     }
+
+    public GLMWeights computeWeightsCOD(double y, double eta, double off, double w, GLMWeights x) {
+      double etaOff = eta + off;
+      x.mu = linkInv(etaOff); // prob estimated response = 1
+      double temp = x.mu*(1-x.mu);
+      x.z = w*(y-x.mu);
+      x.w = w*(temp < 1e-6?1e-6:temp);
```

Comment:
This part could go wrong (concentrating only at numerics). Let's use Math.max, which handles all the edge cases including NaNs correctly.

Reply:
Also, is 1e6 really the desired value returned ?

Reply:
x.mu is a probability from 0 to 1 and x.mu*(1-x.mu) can be small.  Tomas put it in there to make sure the algorithm is numerically table.  I changed it per your suggestion and made more changes to the code to remove all such similar comparisons in GLMModel.java